### DISCUSSION TITLE
Adding support for clearing only failed and skipped tasks

### DISCUSSION THREAD

**Discussion by oz-r** [apache/airflow#28770] [Ideas]
### Description

Airflow now marks running tasks as `skipped` when a DagRun times out. This is great (because historically it did nothing) but it brings up an issue where restarting a DagRun that timed out from the point where it timed out is challenging. You can clear a DagRun right now and you can specify a number of task states to filter (all, failed, running, etc). I'm proposing we add a filter for `skipped` tasks.

An alternative is for DagRun timeouts to mark running tasks as failed.

Aside: It's trivial to do this outside of Airflow by parsing the states of all the tasks and including skipped tasks in a list of tasks to clear. I'm proposing to include it inside of Airflow DagRun clear behavior.

### Use case/motivation

Our users want to be able to easily kickoff timed out DagRuns from the point they timed out. 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Propose PR and let's see how it looks.
